pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -184,6 +210,168 @@ func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Co
+	case configv1.PowerVSPlatformType:
+		return mapi2capi.FromPowerVSMachineAndInfra(mapiMachine, r.Infra).ToMachineAndInfrastructureMachine() //nolint:wrapcheck
+	default:
+		return nil, nil, nil, fmt.Errorf("%w: %s", errPlatformNotSupported, r.Platform)
+	}
+}
+
+// createOrUpdateCAPIInfraMachine creates a CAPI infra machine from a MAPI machine, or updates if it exists and it is out of date.
+func (r *MachineSyncReconciler) createOrUpdateCAPIInfraMachine(ctx context.Context, mapiMachine *machinev1beta1.Machine, infraMachine client.Object, newCAPIInfraMachine client.Object) (ctrl.Result, error) { //nolint:unparam
+	logger := log.FromContext(ctx)
+
+	if infraMachine == nil {
+		if err := r.Create(ctx, newCAPIInfraMachine); err != nil {
+			logger.Error(err, "Failed to create CAPI infra machine")
+			createErr := fmt.Errorf("failed to create CAPI infra machine: %w", err)
+
+			if condErr := r.updateSynchronizedConditionWithPatch(ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToCreateCAPIInfraMachine, createErr.Error(), nil); condErr != nil {
+				return ctrl.Result{}, utilerrors.NewAggregate([]error{createErr, condErr})
+			}
+
+			return ctrl.Result{}, createErr
+		}
+
+		logger.Info("Successfully created CAPI infra machine")
+
+		return ctrl.Result{}, nil
+	}
+
+	isEqualCAPIInfraMachine, err := capiInfraMachineIsEqual(r.Platform, infraMachine, newCAPIInfraMachine)
+	if err != nil {
+		logger.Error(err, "Failed to check CAPI infra machine diff")
+		updateErr := fmt.Errorf("failed to check CAPI infra machine diff: %w", err)
+
+		if condErr := r.updateSynchronizedConditionWithPatch(
+			ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToUpdateCAPIInfraMachine, updateErr.Error(), nil); condErr != nil {
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{updateErr, condErr})
+		}
+
+		return ctrl.Result{}, updateErr
+	}
+
+	if isEqualCAPIInfraMachine {
+		logger.Info("No changes detected in CAPI infra machine")
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("Updating CAPI infra machine")
+
+	if err := r.Update(ctx, newCAPIInfraMachine); err != nil {
+		logger.Error(err, "Failed to update CAPI infra machine")
+
+		updateErr := fmt.Errorf("failed to update CAPI infra machine: %w", err)
+
+		if condErr := r.updateSynchronizedConditionWithPatch(
+			ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToUpdateCAPIInfraMachine, updateErr.Error(), nil); condErr != nil {
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{updateErr, condErr})
+		}
+
+		return ctrl.Result{}, updateErr
+	}
+
+	logger.Info("Successfully updated CAPI infra machine")
+
+	return ctrl.Result{}, nil
+}
+
+// createOrUpdateCAPIMachine creates a CAPI machine from a MAPI one, or updates if it exists and it is out of date.
+func (r *MachineSyncReconciler) createOrUpdateCAPIMachine(ctx context.Context, mapiMachine *machinev1beta1.Machine, capiMachine *capiv1beta1.Machine, newCAPIMachine *capiv1beta1.Machine) (ctrl.Result, error) { //nolint:unparam
```

Comment:
Why unparam?

Reply:
It is for the empty `ctrl.Result{}` that is always returned.
Stylistically I'd prefer to keep that in the function returns, as it is a method over the reconciler.